### PR TITLE
fix: remove LDAP groups claim and store them on server

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: remove LDAP groups claim and store them on server

## Motivation and Context
Groups information shall be now stored as part of the
credential data structure, this is a more idiomatic
way to support large LDAP groups.

Avoids the complication of setups where LDAP groups
can be in the range of 150+ which may lead to excess
HTTP header size > 8KiB, to reduce such an occurrence
we shall save the group information on the server as
part of the credential data structure.

Bonus change support multiple policies across user types

## How to test this PR?
No functionality should change

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by harshavardhana** [comment]
@Alevsk @donatello @vadmeste can I get some reviews?

**Comment by Alevsk** [comment]
> @Alevsk @donatello @vadmeste can I get some reviews?

sure harsha, reviewing it right now

**Review by Alevsk — APPROVED** [review]
LGTM

File: cmd/iam.go [hunk 1]
```diff
@@ -162,16 +162,37 @@ func newGroupInfo(members []string) GroupInfo {
...
+			continue
+		}
+		policies = append(policies, policy)
+	}
+	return policies
 }
 
 func (mp MappedPolicy) policySet() set.StringSet {
-	return set.CreateStringSet(strings.Split(mp.Policy, ",")...)
+	var policies []string
```
Comment (vadmeste) [inline cmd/iam.go:183]: Looks like easier this way:
` return set.CreateStringSet(mp.toSlice()...)`

**Review by vadmeste — COMMENTED** [review]
One comment, LGTM, not tested yet

File: cmd/iam.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/iam.go:183]: It is to avoid the empty value in map as a key @vadmeste which is a real thing btw.

**Review by donatello — APPROVED** [review]
LGTM

**Pull request merged**